src/main/java/org/elasticsearch/index/query/TermQueryParser.java

```diff
@@ -80,27 +74,19 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             }
             parser.nextToken();
         } else {
-            value = parser.text();
+            value = parser.objectText();
```

Comment:
One question I had myself here: I had to change this for the `fromXContentent()" test to work. Otherwise, if the value is e.g. an int, it is written to the XContent query as a number value. With the original `parser.text()` it is then read in as String (which doesn't seem to matter for the later toQuery(), since there BytesRefs.toBytesRef(value) is used, which calls toString() anyway) but blows up the newly introduced equality test. I changes this to `objectText()` because this does some checking based in the JsonToken type.